File: src/libs/sdk/dotnet/Ums.Sdk.Authorization.Testing/TestAuthGraphAccessor.cs
using Ums.Sdk.Contracts;

namespace Ums.Sdk.Authorization.Testing;

/// <summary>
/// Trivial <see cref="IAuthGraphAccessor"/> backed by a single in-memory graph.
/// Useful for unit tests that wire the validator directly without DI.
/// </summary>
public sealed class TestAuthGraphAccessor : IAuthGraphAccessor
{
    public AuthorizationGraph? Current { get; set; }

    public TestAuthGraphAccessor() { }
    public TestAuthGraphAccessor(AuthorizationGraph? graph) { Current = graph; }
}

File: src/libs/sdk/typescript/sdk-testing/src/test-auth-graph-accessor.ts
import type { AuthorizationGraph } from '@ums/sdk-contracts';
import type { AuthGraphAccessor } from '@ums/sdk-authorization';

/**
 * Trivial accessor backed by a single in-memory graph. Useful in unit tests that wire the
 * validator directly without any framework integration.
 */
export class TestAuthGraphAccessor implements AuthGraphAccessor {
  private graph: AuthorizationGraph | null;

  constructor(graph: AuthorizationGraph | null = null) {
    this.graph = graph;
  }

  current(): AuthorizationGraph | null {
    return this.graph;
  }

  set(graph: AuthorizationGraph | null): void {
    this.graph = graph;
  }

  clear(): void {
    this.graph = null;
  }
}
